extmod/modbluetooth.c

```diff
@@ -959,11 +961,166 @@ STATIC mp_obj_t bluetooth_ble_invoke_irq(mp_obj_t none_in) {
     return mp_const_none;
 }
 STATIC MP_DEFINE_CONST_FUN_OBJ_1(bluetooth_ble_invoke_irq_obj, bluetooth_ble_invoke_irq);
+#endif // !MICROPY_PY_BLUETOOTH_USE_SYNC_EVENTS
 
 // ----------------------------------------------------------------------------
 // Port API
 // ----------------------------------------------------------------------------
 
+#if MICROPY_PY_BLUETOOTH_USE_SYNC_EVENTS
+
+STATIC mp_obj_t invoke_irq_handler(uint16_t event,
+    const uint16_t *u16, size_t n_u16,
+    const uint8_t *u8, size_t n_u8,
+    const uint8_t *addr,
+    const int8_t *i8, size_t n_i8,
+    const mp_obj_bluetooth_uuid_t *uuid,
+    const uint8_t *data, size_t data_len) {
+    mp_obj_bluetooth_ble_t *o = MP_OBJ_TO_PTR(MP_STATE_VM(bluetooth));
+    if (o->irq_handler == mp_const_none) {
+        return mp_const_none;
+    }
+
+    mp_obj_array_t mv_addr;
+    mp_obj_array_t mv_data;
+
+    mp_obj_tuple_t *data_tuple = mp_local_alloc(sizeof(mp_obj_tuple_t) + sizeof(mp_obj_t) * MICROPY_PY_BLUETOOTH_MAX_EVENT_DATA_TUPLE_LEN);
+    data_tuple->base.type = &mp_type_tuple;
+    data_tuple->len = 0;
+
+    for (size_t i = 0; i < n_u16; ++i) {
+        data_tuple->items[data_tuple->len++] = MP_OBJ_NEW_SMALL_INT(u16[i]);
+    }
+    if (n_u8) {
```

Comment:
Would you be able to make this `for (size_t i = 0; i < n_u8; ++i) {` matching the `n_u16` loop above to cover any (GATTS_ENC_UPDATE) events that have multiple u8's?

The matching change would also be needed in `void ringbuf_extract()`

Reply:
Sure thing, done.